removes period from end of folder names for linux

yobleck opened the pull request:
fixes error where folder can not be downloaded because name of folder ends with period "."

EDIT:
Resolves #1385

yobleck committed:
removes period from end of folder names for linux

fixes error where folder can not be downloaded because name ends with period

yobleck committed:
fixed typos

27136773 commented on src/web/mjs/engine/Storage.mjs:
```diff
@@ -802,6 +802,9 @@ export default class Storage {
             return path.replace( /\.+$/g, '' ).trim(); // remove trailing dots and whitespaces
         }
         if( this.platform.indexOf( 'linux' ) === 0 ) {
+            if(path.substring(path.length-1,path.length)==".") { //check if last character is a period
```
This seems to be not robust enough, what if there are 2 or more dots at the end
Maybe use a regex instead, something like this:
`path.replace(/[\.\s]+$/, '')`

27136773 commented on src/web/mjs/engine/Storage.mjs:
```diff
@@ -802,6 +802,9 @@ export default class Storage {
             return path.replace( /\.+$/g, '' ).trim(); // remove trailing dots and whitespaces
         }
         if( this.platform.indexOf( 'linux' ) === 0 ) {
```
Should this improvement be done for all operating systems, not just linux?

yobleck committed:
strips periods and spaces from end of folder name

path = path.replace( /[/\s\.]+$/g, '' ); // remove trailing dots and whitespaces
added on lines 805 and 810

(also fixes typo sanatize -> sanitize)

yobleck committed:
removes unnecessary "\"

yobleck committed:
Update Storage.mjs

yobleck committed:
Add files via upload

yobleck committed:
switched to windows eol

yobleck committed:
Add files via upload

yobleck committed:
Add files via upload

27136773 commented on src/web/mjs/engine/Storage.mjs:
```diff
@@ -795,17 +795,19 @@ export default class Storage {
      * WINDOWS: wxT("\\/:*?\"<>|\r\n\t");
      */
-    sanatizePath ( path ) {
+    sanitizePath ( path ) {
```
This is a public method, if renamed all its occurrences must be renamed as well, otherwise the application stops to work correctly.

See:
- BookmarkConnector.mjs
- Manga.mjs
- Chapter.mjs

yobleck committed:
sanatize -> sanitize

yobleck committed:
revert sana to sani

causes errors that I'm too inexperienced to track down
